docs/CONTRIBUTING.rst

```diff
@@ -0,0 +1,148 @@
+
+Do you have questions about HDMF-ZARR?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+See our `hdmf-dev.github.io <https://hdmf-dev.github.io/>`_ website for details.
+
+Informal discussions between developers and users?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The https://nwb-users.slack.com slack is currently used for informal discussions between developers and users.
+
+.. _sec-contributing:
+
+Contributing Patches and Changes
+--------------------------------
+
+To contribute to HDMF-ZARR you must submit your changes to the ``dev`` branch via a `Pull Request <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request>`_.
+
+From your local copy directory, use the following commands.
+
+1) First create a new branch to work on
+
+.. code-block:: bash
+
+    $ git checkout -b <new_branch>
+
+2) Make your changes.
+
+3) Push your feature branch to origin (i.e. GitHub)
+
+.. code-block:: bash
+
+    $ git push origin <new_branch>
+
+4) Once you have tested and finalized your changes, create a pull request targeting ``dev`` as the base branch. Be sure to use our `pull request template <https://github.com/hdmf-dev/hdmf-zarr/blob/dev/.github/pull_request_template.md>`_ and:
+
+    * Ensure the PR description clearly describes the problem and solution.
+    * Include the relevant issue number if applicable.
+    * Before submitting, please ensure that:
+      * The proposed changes include an addition to ``CHANGELOG.md`` describing your changes. To label the change with the PR number, you will have to first create the PR, then edit the ``CHANGELOG.md`` with the PR number, and push that change.
+      * The code follows our coding style. This can be checked running ``ruff`` from the source directory.
+    * **NOTE:** Contributed branches will be removed by the development team after the merge is complete and should, hence, not be used after the pull request is complete.
+
+.. _sec-styleguides:
+
+Style Guides
+------------
+
+Python Code Style Guide
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Before you create a Pull Request, make sure you are following the PEP8_ style guide. .
+To check whether your code conforms to the HDMF-ZARR style guide, simply run the ruff_ tool in the project's root
+directory. ``ruff`` will also sort imports automatically and check against additional code style rules.
+
+We also use ``ruff`` to sort python imports automatically and double-check that the codebase
```

Comment:
This isn't quite accurate yet, but it's fine to put it here. Can you create a ticket to adopt ruff and the other modern tooling that is now in HDMF?